content/en/platform/corda/5.2/deploying-operating/deployment/upgrading521/_index.md

```diff
@@ -0,0 +1,517 @@
+
+A simple port forward to Kafka is not sufficient as this only forwards traffic to the bootstrap server. The redirection from the bootstrap to a Kafka broker fails unless you are on the same subnet as the Kafka broker’s advertised listener name. Additionally, if SSL is enabled for your Kafka deployment, you cannot access it at a different host name from those covered by the certificate.
+
+This guide cannot describe how to create a connection to any arbitrary Kafka deployment. It is up to the upgrader to ensure this is possible from their host. The examples in the guide use local host DNS mapping to connect to a cluster. The host is configured so that a known advertised listener name maps to an IP address by manipulating the local host's file. However, this might not be appropriate for your deployment. For example, the automated pipelines at R3 use a pod from within the Corda cluster to perform Kafka upgrade steps. Configuring such a pod, if it is required, is the responsibility of the Corda Operator.
+
+You can test Kafka access using off-the-shelf Kafka tools. For example, if you can list Kafka topics using a command similar to this, you have the required access from your host:
+
+```
+kafka-topics --bootstrap-server=prereqs-kafka.test-namespace:9092 --list
+```
+
+### REST API Access
+
+To get information about TLS certificate aliases, the data migration step requires access to the Corda REST API. This is required at the same stage when data is extracted from Kafka, which means access to Kafka and the Corda REST API must be available in the same environment at the same time. You must pass credentials and locations of both to the Corda CLI plugin when executed.
+
+Also, you must have `GET` access to the `/key` and `/certificate` APIs.
+
+### Corda CLI Tool and Corda 5 Plugins
+
+Some upgrade steps use the {{< tooltip >}}Corda CLI{{< /tooltip >}} tool used also in Corda 5 [manual bootstrapping]({{< relref "../deploying/manual-bootstrapping.md" >}}). Users who entirely bootstrapped Corda 5 using the Corda Helm chart may not be familiar with this tool because the Corda Helm chart executes it for you. For instructions on how to install Corda CLI, go to [Installing the Corda CLI]({{< relref "../../../application-networks/tooling/installing-corda-cli.md" >}}).
+
+The Corda CLI tool and plugins are also published internally as Docker images. The examples in this script use those Docker images so that:
+* The Corda CLI tool does not need to be downloaded or installed.
+* The latest published tool is always available.
+* The correct version is always used.
+
+{{< note >}}
+The version of the Corda CLI tool and plugins must match the version of Corda 5 being upgraded to, that is, 5.2.1.
+{{< /note >}}
+
+The functionality of the tools, the database schemas and Kafka topics embedded in them pertain to the version of Corda they were built against.
+
+If you are not using Docker images, where you come across an example of the form:
```

Comment:
Lines 85 to 95 inclusive must be removed. Customers will not use the docker images. If you look at https://anna.preview.docs.r3.com/en/platform/corda/5.2/deploying-operating/deployment/upgrading.html for example, all instances of `docker run...` have been replaced with `corda-cli.sh`, we should do that here too.